plugin/SqueezeESP32/Graphics.pm

```diff
@@ -106,6 +120,19 @@ sub displayHeight {
 	return 32;
 }
 
+sub sanitizeSpectrum {
+	my ($spectrum) = shift;
+
+	$spectrum->{small} ||= dclone($SPECTRUM_DEFAULTS{small});
+	$spectrum->{small}->{size} ||= $SPECTRUM_DEFAULTS{small}->{size};
+	$spectrum->{small}->{band} ||= $SPECTRUM_DEFAULTS{small}->{band};
```

Comment:
I'm a bit confused - isn't dclone supposed to do the job here?

Reply:
The `clone` will only set `->{small}` if it didn't exist yet (`||=`). But its existence doesn't mean its properties are defined. Therefore I do initialise them separately - if needed.

Reply:
I need to revise dclone then. I thought it's objective was copying a whole hierarchy. I thought that if there is no {small} key, then it, and all its children would be copied.  If there is one, then nothing happens

Reply:
Exactly. But if `->{small}` was eg. an empty object (`{}`), then `dclone` wouldn't help. In that case the other two lines will initialise those properties.

Reply:
I'm not trying to be picky but then why not just simply leaving the small & band key assignments then? The small key is created if it does not exist.

Reply:
`1/$spectrum->{small}->{size}` would fail if `$spectrum->{small} == {}`.

Reply:
Is this dependent on a Perl version? I just did a quick test and it seems to me that whether small is set, not set, or empty it does not fail

Reply:
if `$spectrum->{small}->{size}` is undefined, then `1/undefined` would certainly fail?

Reply:
A attempt to read to a child will fail (can't use undef as a reference) if the parent key is undefined, but I think a write or an ||= will not fail if the parent is not defined. I mean this example seems to work
```
use strict;

use Data::Dumper;

my $a;

print(Dumper($a));
$a->{small}->{size} ||= 1;
$a->{small}->{band} ||= 2;
print(Dumper($a));

my $a;

$a->{small} = {};
print(Dumper($a));
$a->{small}->{size} ||= 1;
$a->{small}->{band} ||= 2;
print(Dumper($a));
```
But maybe this is unproper Perl use, I don't know.

Reply:
Maybe my suggested solution is overcautious indeed - I don't know the history of that pref. But the case of the `artwork` pref has shown that there can be unexpected side-effects, in particular when changing a scalar pref to some hash or list.

I've come to the conclusion that using a hash in a pref isn't that good an idea. Many mechanisms of the prefs handling don't work correctly. And it's a ref: you can't simply change `$spectrum` without at the same time changing the prefs data. It can be fine if you're aware of the limitations, but it's much easier to break.

Reply:
I agree and I had a few surprises with that. Now that I know, it's fine but I missed it at the beginning.